core/src/main/scala/kafka/log/Log.scala

```diff
@@ -69,10 +72,7 @@ case class LogAppendInfo(var firstOffset: Long,
                          shallowCount: Int,
                          validBytes: Int,
                          offsetsMonotonic: Boolean,
-                         pid: Long,
-                         epoch: Short,
-                         firstSequence: Int,
-                         lastSequence: Int)
+                         pidEntryMap : Map[Long, (PidEntry, PidEntry)])
```

Comment:
Maybe we could create a case class for the tuple of `PidEntry`. Having names will make it easier to follow the code.